src/node/rpc/node_frontend.h

```diff
@@ -357,6 +364,12 @@ namespace ccf
             "No service is available to accept new node.");
         }
 
+        auto config = args.tx.ro(network.config);
+        auto service_config = config->get();
+        auto reconfiguration_type = ReconfigurationType::ONE_TRANSACTION;
+        if (service_config->reconfiguration_type.has_value())
+          reconfiguration_type = service_config->reconfiguration_type.value();
```

Comment:
This seems to be called twice in the `/node/join` endpoint. Could we only call it once and pass the `reconfiguration_type` result around instead?